### PR TITLE
Enum - Windowflags conversion

### PR DESCRIPTION
**Pull request by niceeffort**
Converting enum flags to new style per #1757

### CODE REVIEW

File: src/OpenLoco/src/Ui/WindowManager.cpp [hunk 1]
```diff
@@ -556,11 +556,11 @@ namespace OpenLoco::Ui::WindowManager
         // Border flash invalidation
         for (Ui::Window* w = _windowsEnd - 1; w >= _windows; w--)
         {
-            if ((w->flags & WindowFlags::whiteBorderMask) != 0)
+            if (w->hasFlags(WindowFlags::whiteBorderMask))
             {
                 // TODO: Replace with countdown
-                w->flags -= WindowFlags::whiteBorderOne;
-                if ((w->flags & WindowFlags::whiteBorderMask) == 0)
+                w->flags &~ WindowFlags::whiteBorderOne;
```
Comment (ZehMatt) [inline src/OpenLoco/src/Ui/WindowManager.cpp:562]: This looks wrong, you probably want:
```suggestion
                w->flags &= ~WindowFlags::whiteBorderOne;
```

File: src/OpenLoco/src/Ui/WindowManager.cpp [hunk 2]
```diff
@@ -1662,7 +1662,7 @@ namespace OpenLoco::Ui::WindowManager
     {
         for (auto w = window; w < _windowsEnd; w++)
         {
-            if ((w->flags & WindowFlags::transparent) == 0)
+            if (!w->hasFlags(WindowFlags::transparent))
```
Comment (ZehMatt) [inline src/OpenLoco/src/Ui/WindowManager.cpp:1665]:
```suggestion
            if (!w->isTranslucent())
```

**Review by ZehMatt — CHANGES_REQUESTED** [review]

File: src/OpenLoco/src/Ui/WindowManager.cpp [hunk 3]
```diff
@@ -556,11 +556,11 @@ namespace OpenLoco::Ui::WindowManager
         // Border flash invalidation
         for (Ui::Window* w = _windowsEnd - 1; w >= _windows; w--)
         {
-            if ((w->flags & WindowFlags::whiteBorderMask) != 0)
+            if (w->hasFlags(WindowFlags::whiteBorderMask))
             {
                 // TODO: Replace with countdown
-                w->flags -= WindowFlags::whiteBorderOne;
-                if ((w->flags & WindowFlags::whiteBorderMask) == 0)
+                w->flags &= ~WindowFlags::whiteBorderOne;
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/Ui/WindowManager.cpp:562]: Argh I didn't read this code it was meant to do a -=. Can you change it to the following:
```cpp
if (w->hasFlags(WindowFlags::whiteBorderMask)){

    // TODO: Replace with seperate variable countdown
    // Countdown is stuffed into WindowFlags extract it out and decrement by 1
    const auto newCount = enumValue(w->flags & WindowFlags::whiteBorderMask) - enumValue(WindowFlags::whiteBorderOne);
    // Stuff the new count back into the WindowFlags
    w->flags &= ~WindowFlags::whiteBorderMask;
    w->flags |= static_cast<WindowFlags>(newCount);

    if (!w->hasFlags(WindowFlags::whiteBorderMask)){

Comment (ZehMatt) [reply src/OpenLoco/src/Ui/WindowManager.cpp:562]: I see, we should maybe only allow windows flags access by getter/setter for as long the count is also encoded in there, should also make this a bit more readable again. Not required for this PR

**Review by duncanspumpkin — APPROVED** [review]

**Comment by AaronVanGeffen** [comment]
There's a regression from this PR; the outer border of all windows looks different now.

**Pull request merged**